public/jira/rollup/dates/dates.ts

```diff
@@ -0,0 +1,213 @@
+import { DueData, StartData } from "../../../shared/issue-data/date-data";
+import {
+  rollupGroupedHierarchy,
+  groupIssuesByHierarchyLevelOrType,
+  zipRollupDataOntoGroupedData,
+  IssueOrRelease,
+  RollupResponse,
+  ReportingHierarchyIssueOrRelease,
+} from "../rollup";
+import {
+  selectDue,
+  selectStart,
+  sortByDue,
+  sortByStart,
+} from "../../shared/helpers";
+import { DerivedIssue } from "../../derived/derive";
+
+const methods = {
+  parentFirstThenChildren,
+  childrenOnly,
+  childrenFirstThenParent,
+  widestRange,
+  parentOnly,
+};
+export type CalculationName = keyof typeof methods;
+
+export type RollupDateData = Partial<StartData & DueData>;
+
+export type RolledupDatesReleaseOrIssue = IssueOrRelease<DerivedIssue> & {
+  rollupDates: RollupDateData;
+};
+
+const getStartData = (d: Partial<StartData> | null) => {
+  if(!d)return {}
+  
+  const {start,startFrom}=d
+
+  return ({
+    ...(start && { start }),
+    ...(startFrom && { startFrom }),
+  });
+};
+const getDueData = (d: Partial<DueData> | null) => {
+  if(!d)return {}
+  
+  const {due,dueTo}=d
+
+  return ({
+    ...(due && { due }),
+    ...(dueTo && { dueTo }),
```

Comment:
this was the best solution i could come up with for copying the date data with no change to the expected results. leme know if there's a better way to do it in TS

Reply:
I'm not really sure that this function is trying to do, if you're just trying to copy the object, you could use `structuredClone` instead. 

If you do keep this function can you change the arg name and add an explicit return type. I _think_ this is returning

```ts
{
    dueTo?: {
        message: string;
        reference: NormalizedIssue;
    } | undefined;
    due?: Date | undefined;
}
```
but is that the intention?

Reply:
so as far as i can tell, the intention behind getStartData and getDueData is to copy only the props related to the specific type, StartData or DueData. They're eventually getting recombined later, but they have to be split out from a common object and processed separately first. Both spread ( {...} ) and structuredClone return the entire object, including the props we don't want. This was causing the tests to fail cuz one or the other would end up overwriting the final value when recombined